dc-cudami-editor/src/components/Autocomplete.jsx

```diff
@@ -11,146 +11,148 @@ import InputWithSpinner from './InputWithSpinner'
 import PreviewImage from './PreviewImage'
 import {getLabelValue} from './utils'
 
-class Autocomplete extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      loading: false,
-      /* defines the number of suggestions to be fetched */
-      maxElements: props.maxElements ?? 25,
-      searchTerm: '',
-      suggestions: [],
-      totalElements: 0,
-    }
-  }
-
-  getSuggestionAsString = ({label}) =>
-    `${getLabelValue(
-      label,
-      this.props.activeLanguage,
-      this.context.defaultLanguage,
-    )}`
-
-  onChange = (_, {newValue: searchTerm}) => {
-    this.setState({
-      searchTerm,
-    })
-  }
-
-  onSuggestionsClearRequested = () => {
-    this.setState({
-      suggestions: [],
-      totalElements: 0,
-    })
-  }
-
-  onSuggestionsFetchRequested = async ({value: searchTerm}) => {
-    if (searchTerm.length < 2) {
-      return
-    }
-    this.setState({loading: true})
-    const {content: suggestions, totalElements} = await this.props.search(
-      this.context.apiContextPath,
-      searchTerm,
-      0,
-      this.state.maxElements,
-    )
-    this.setState({
-      loading: false,
-      suggestions,
-      totalElements,
-    })
-  }
-
-  onSuggestionSelected = (_, {suggestion}) => {
-    this.setState({
-      searchTerm: '',
-    })
-    this.props.onSelect(suggestion)
-  }
-
-  renderInputComponent = (inputProps) => {
-    return (
-      <FormGroup className="mb-0">
-        <InputWithSpinner
-          inputProps={inputProps}
-          loading={this.state.loading}
-        />
-      </FormGroup>
-    )
-  }
-
-  renderSuggestion = ({label, previewImage, previewImageRenderingHints}) => {
-    return (
-      <Row>
-        <Col md="1">
-          <PreviewImage
-            image={previewImage}
-            renderingHints={previewImageRenderingHints}
-            width={50}
-          />
-        </Col>
-        <Col className="text-left" md="11">
-          {getLabelValue(
-            label,
-            this.props.activeLanguage,
-            this.context.defaultLanguage,
-          )}
-        </Col>
-      </Row>
-    )
-  }
+const renderInputComponent = (inputProps, loading) => (
```

Comment:
Das ist ja eigentlich keine `render` Funktion sondern eine eigenständige Komponente, oder? Also vom Namen her eher `InputComponent`? Das `loading` könntest du dann auch direkt in die `inputProps` mit reinnehmen, oder?

Reply:
Das `loading` würde ich nicht so gerne in die `inputProps` mit reinnehmen, da es nicht direkt was mit dem Input zu tun hat :thinking:

Reply:
Musst ja nicht?

```
<InputComponent inputProps={inputProps} loading={loading} />
```

```
const InputComponent = ({ inputProps, loading}) => ( /* ... */);
```

Reply:
Hatte den Kontext nicht im Blick, passt